custom_components/tuya_local/devices/ezaiot_r9lite_thermostat.yaml

```diff
@@ -0,0 +1,219 @@
+name: EZAIOT R9Lite thermostat
+#https://www.aliexpress.com/item/1005006020556081.html  Electric Heating 16A version
+products:
+  - id: dq6nlukkifyawj9n
+    name: EZAIOT R9Lite thermostat
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: 'off'
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: auto
+              value: auto
+            - dps_val: manual
+              value: heat
+    - id: 2
+      type: string
+      name: mode
+      hidden: true
+    - id: 16
+      type: integer
+      name: temperature
+      range:
+        min: 50
+        max: 10000
+      mapping:
+        - scale: 10
+          step: 5
+          constraint: temperature_unit
+          conditions:
+            - dps_val: f
+              step: 10
+    - id: 19
+      type: integer
+      name: max_temperature
+      mapping:
+        - scale: 10
+    - id: 23
+      type: string
+      name: temperature_unit
+      mapping:
+        - dps_val: c
+          value: C
+        - dps_val: f
+          value: F
+    - id: 24
+      type: integer
+      name: current_temperature
+      mapping:
+        - scale: 10
+    - id: 26
+      type: integer
+      name: min_temperature
+      mapping:
+        - scale: 10
+    - id: 101
+      type: integer
+      name: unknown_101
+    - id: 112
+      type: integer
+      name: unknown_112
+    - id: 102
+      type: integer
+      name: Temperature adjust step
+    - id: 110
+      type: boolean
+      name: Valve state boolean
+    - id: 105
+      type: string
+      name: Device name
+    - id: 108
+      type: integer
+      name: Temperature calibration step
+secondary_entities:
+  - entity: switch
+    translation_key: anti_frost
+    category: config
+    dps:
+      - id: 10
+        type: boolean
+        name: switch
+  - entity: select
+    translation_key: temperature_unit
+    category: config
+    dps:
+      - id: 23
+        type: string
+        name: option
+        mapping:
+          - dps_val: c
+            value: celsius
+          - dps_val: f
+            value: fahrenheit
+  - entity: select
+    name: Schedule
+    category: config
+    icon: 'mdi:calendar-clock'
+    dps:
+      - id: 31
+        type: string
+        name: option
+        mapping:
+          - dps_val: '5_2'
+            value: Weekday+Weekend
+          - dps_val: '6_1'
+            value: Mon-Sat+Sun
+          - dps_val: '7'
+            value: Daily
+  - entity: binary_sensor
+    name: Valve
+    class: opening
```

Comment:
Valid classes for valve are "water" and "gas".

Reply:
It's electric heating thermostat Classes "water" and "gas" are wrong. Additionally please see https://github.com/make-all/tuya-local/blob/2a99434d28b1282583f0fd731209e3aa6932c723/custom_components/tuya_local/devices/beok_tr9b_thermostat.yaml#L101. There is Valve with class "opening" too.

Reply:
There is? It appears to be missing from the [documentation](https://www.home-assistant.io/integrations/valve/#device-class), so probably it is safer not to use it in case it is suddenly removed.

If the valve is not for water or gas to pass through, then there should be no class.

Reply:
In fact dps id 36 (valve state- string represenation) is redundant with dps id 110 (valve state- boolean representation) so I removed sensor for dps id 36 and create as climate attribute and mark the attribute as hidden. Please see https://github.com/janouskovec/tuya-local/commit/1aaad940762c6e28c690c633e04ae04ae2be06c5. So this issue is solved.